### PR TITLE
Add support for SAML2 authentication

### PR DESCRIPTION
**Pull request by zackgalbreath**

### CODE REVIEW

**Review by josephsnyder — APPROVED** [review]
Everything looks solid.  The Docker-compose system has no problem with the new artisan packages, but I can't quite test the `saml2_tenant` creation.  

The only non-blocking nitpick would be that some of the functions still put the return type in the doc comment instead of in the function definition while others do not:

- `saml2Login` in `LoginController.php`
- `create` in `RegisterController.php`
- `shouldDiscoverEvents` in `EventServiceProvider.php`

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -127,4 +129,27 @@ public static function staticShowLoginForm()
                 'js_version' => self::getJsVersion()
             ]);
     }
+
+    public static function saml2Login() : \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
```
Comment (williamjallen) [inline app/Http/Controllers/Auth/LoginController.php:133]: The return type can be cleaned up by importing the relevant class.  `\Illuminate\Http\Response` is already being used, and `\Illuminate\Http\RedirectResponse` can be added.  The end result would be a return type of `Response|RedirectResponse` which is a little bit easier to read.

File: phpstan.neon [hunk 2]
```diff
@@ -30,9 +30,4 @@ parameters:
     level: 6
 
 includes:
-    - vendor/nunomaduro/larastan/extension.neon
-    - vendor/phpstan/phpstan-strict-rules/rules.neon
-    - vendor/phpstan/phpstan-deprecation-rules/rules.neon
-    - vendor/phpstan/phpstan-phpunit/extension.neon
-    - vendor/phpstan/phpstan-phpunit/rules.neon
```
Comment (williamjallen) [inline phpstan.neon:37]: Were these deliberately removed?

File: composer.json [hunk 3]
```diff
@@ -47,8 +48,10 @@
     "fakerphp/faker": "^1.21",
     "mockery/mockery": "^1.0",
     "nunomaduro/larastan": "^2.4",
+    "phpstan/extension-installer": "^1.3",
     "phpstan/phpstan": "^1.10",
     "phpstan/phpstan-deprecation-rules": "^1.1",
+    "phpstan/phpstan-mockery": "^1.1",
```
Comment (williamjallen) [inline composer.json:54]: These are necessarily related to the rest of this PR but it's probably alright to include them anyway

**Review by williamjallen — COMMENTED** [review]
Overall, code looks decent.  A few minor nitpicks noted below and then I think it should be ready to go.

**Pull request merged**